### PR TITLE
feat: add api calls to get categories and last comment

### PR DESCRIPTION
**Pull request by jnduli**
# Description
Added api calls to:

- get list of categories
-  get list of species
- get the last comment related to the version

## Testing
```
GET http://127.0.0.1:9094/api/metadata/wiki/12 HTTP/1.1
Content-Type: application/json

# Result
HTTP/1.0 200 OK
Content-Type: application/json
Content-Length: 278
Server: Werkzeug/2.0.2 Python/3.10.7
Date: Mon, 26 Aug 2024 17:42:40 GMT


{
  "categories": [
    "Behavior", 
    "Biochemistry"
  ], 
  "comment": "XXXX", 
  "email": "XXXX", 
  "initial": "XXXX", 
  "pubmed_ids": [
    "XXXX", 
    "XXXX"
  ], 
  "reason": "XXXX", 
  "species": "mouse", 
  "symbol": "XXXX", 
  "version": 14, 
  "weburl": "XXXX"
}

## API for species
GET http://127.0.0.1:9094/api/metadata/wiki/species HTTP/1.1
Content-Type: application/json

HTTP/1.0 200 OK
Content-Type: application/json
Content-Length: 366
Server: Werkzeug/2.0.2 Python/3.10.7
Date: Mon, 26 Aug 2024 17:43:11 GMT


{
  "Oryzias latipes": "Oryzias latipes (Japanese medaka)", 
  "arabidopsis": "Arabidopsis thaliana", 
  "barley": "Barley", 
  "bat": "Bat (Glossophaga soricina)", 
  "drosophila": "Drosophila", 
  "human": "Human", 
  "macaque monkey": "Macaque monkey", 
  "mouse": "Mouse", 
  "poplar": "Poplar", 
  "rat": "Rat", 
  "soybean": "Soybean", 
  "tomato": "Tomato"
}



```

### CODE REVIEW

File: gn3/api/metadata_api/wiki.py [hunk 1]
```diff
@@ -61,3 +61,28 @@ def edit_wiki(comment_id: int):
             )
         return jsonify({"success": "ok"})
     return jsonify(error="Error editting wiki entry, most likely due to DB error!"), 500
+
+
+@wiki_blueprint.route("/<int:comment_id>", methods=["GET"])
+def get_wiki(comment_id: int):
+    with db_utils.database_connection(current_app.config["SQL_URI"]) as conn:
+        return jsonify(wiki.get_latest_comment(conn, comment_id))
+    return jsonify(error="Error editting wiki entry, most likely due to DB error!"), 500
```
Comment (BonfaceKilz) [inline gn3/api/metadata_api/wiki.py]: *issue:* This comment is mis-leading; Please update it.

File: gn3/api/metadata_api/wiki.py [hunk 2]
```diff
@@ -61,3 +61,28 @@ def edit_wiki(comment_id: int):
...
+
+
+@wiki_blueprint.route("/categories", methods=["GET"])
+def get_categories():
+    with db_utils.database_connection(current_app.config["SQL_URI"]) as conn:
+        cursor = conn.cursor()
+        categories_dict = wiki.get_categories(cursor)
+        return jsonify(categories_dict)
+    return jsonify(error="Error editting wiki entry, most likely due to DB error!"), 500
```
Comment (BonfaceKilz) [inline gn3/api/metadata_api/wiki.py]: *issue:* This comment is mis-leading.  Please update it.

File: gn3/api/metadata_api/wiki.py [hunk 3]
```diff
@@ -61,3 +61,28 @@ def edit_wiki(comment_id: int):
...
+
+
+@wiki_blueprint.route("/species", methods=["GET"])
+def get_species():
+    with db_utils.database_connection(current_app.config["SQL_URI"]) as conn:
+        cursor = conn.cursor()
+        species_dict = wiki.get_species(cursor)
+        return jsonify(species_dict)
+    return jsonify(error="Error editting wiki entry, most likely due to DB error!"), 500
```
Comment (BonfaceKilz) [inline gn3/api/metadata_api/wiki.py]: *issue:* Ditto.

File: gn3/db/wiki.py [hunk 4]
```diff
@@ -1,12 +1,38 @@
...
+
+from MySQLdb.cursors import DictCursor
 
 
 class MissingDBDataException(Exception):
     """Error due to DB missing some data"""
 
 
+def get_latest_comment(connection, comment_id: str) -> int:
+    cursor = connection.cursor(DictCursor)
```
Comment (BonfaceKilz) [inline gn3/db/wiki.py]: *issue:* I'm sure pylint will complain about a missing docstring here.

**Review by BonfaceKilz — CHANGES_REQUESTED** [review]
For `wiki.py` in this PR, pylint will complain about missing doc-strings.

File: gn3/db/wiki.py (hunk 4 above)
Comment (jnduli) [reply gn3/db/wiki.py]: Fixed

File: gn3/api/metadata_api/wiki.py (hunk 3 above)
Comment (jnduli) [reply gn3/api/metadata_api/wiki.py]: Fixed

File: gn3/api/metadata_api/wiki.py (hunk 2 above)
Comment (jnduli) [reply gn3/api/metadata_api/wiki.py]: Fixed

File: gn3/api/metadata_api/wiki.py (hunk 1 above)
Comment (jnduli) [reply gn3/api/metadata_api/wiki.py]: Fixed

**Review by BonfaceKilz — APPROVED** [review]

**Pull request merged**